### PR TITLE
OG-788: Blacklist for Relay Servers in client configuration

### PR DESCRIPTION
**Pull request by forshtat**

### CODE REVIEW

File: packages/common/src/ConfigResponse.ts [hunk 1]
```diff
@@ -21,8 +21,14 @@ export interface ConfigEntry {
   gsnConfig: Partial<GSNConfig>
 }
 
+export interface BlacklistedRelayServerInfo {
```
Comment (drortirosh) [inline packages/common/src/ConfigResponse.ts:24]: I think it should be a `string[]`, not some compex structure, just like the preferred relays

Comment (forshtat) [reply packages/common/src/ConfigResponse.ts:24]: Don't you think we should differentiate between manager address and host? Just drop everything in one string array?

Comment (forshtat) [reply packages/common/src/ConfigResponse.ts]: Fixed.

File: packages/provider/src/KnownRelaysManager.ts [hunk 2]
```diff
@@ -80,7 +80,16 @@ export class KnownRelaysManager {
 
     this.logger.info(`fetchRelaysAdded: found ${relayInfos.length} relays`)
 
-    const filteredRelayInfos = relayInfos.filter(this.relayFilter)
+    const blacklistFilteredRelayInfos = relayInfos.filter((info: RelayRegisteredEventInfo) => {
+      let host = ''
+      try {
+        host = new URL(info.relayUrl).host.toLowerCase()
+      } catch (e) {}
+      const isHostBlacklisted = this.config.blacklistedRelays.find(relay => host.includes(relay.toLowerCase())) != null
```
Comment (drortirosh) [inline packages/provider/src/KnownRelaysManager.ts:88]: better match against relayUrl, and not just the "host" part.
that is, allow filtering by full url, including the "https" prefix
(e.g. copying a URL from the preferred relay into the blacklistedRelays should work)

not sure if the blacklist-by-address is useful, but it causes no harm either.

Comment (drortirosh) [reply packages/provider/src/KnownRelaysManager.ts:88]:
```js
 const isHostBlacklisted = this.config.blacklistedRelays.find(relay => info.relayUrl.toLowerCase().includes(relay.toLowerCase())) != null
```

**Pull request merged**